hoodie-common/src/main/java/com/uber/hoodie/common/table/log/HoodieCompactedLogRecordScanner.java

```diff
@@ -78,24 +78,31 @@
   private String payloadClassFQN;
   // Store only the last log blocks (needed to implement rollback)
   Deque<HoodieLogBlock> lastBlocks = new ArrayDeque<>();
+  // Maximum size of map holding all records for this fileSlice=
+  private long maxMemorySizeInBytes;
+
 
   public HoodieCompactedLogRecordScanner(FileSystem fs, String basePath, List<String> logFilePaths,
-      Schema readerSchema, String latestInstantTime) {
+                                         Schema readerSchema, String latestInstantTime, Long maxMemorySizeInBytes) {
     this.readerSchema = readerSchema;
     this.latestInstantTime = latestInstantTime;
     this.hoodieTableMetaClient = new HoodieTableMetaClient(fs, basePath);
     // load class from the payload fully qualified class name
     this.payloadClassFQN = this.hoodieTableMetaClient.getTableConfig().getPayloadClass();
+    this.maxMemorySizeInBytes = maxMemorySizeInBytes;
 
-    // Store merged records for all versions for this log file
-    Map<String, HoodieRecord<? extends HoodieRecordPayload>> records = Maps.newHashMap();
-    // iterate over the paths
-    Iterator<String> logFilePathsItr = logFilePaths.iterator();
-    while (logFilePathsItr.hasNext()) {
-      HoodieLogFile logFile = new HoodieLogFile(new Path(logFilePathsItr.next()));
-      log.info("Scanning log file " + logFile.getPath());
-      totalLogFiles.incrementAndGet();
-      try {
+    ExternalSpillableMap<String, HoodieRecord<? extends HoodieRecordPayload>> records;
+    try {
+      // Store merged records for all versions for this log file, set the maxInMemoryMapSize to half,
+      // assign other half to the temporary map needed to read next block
+      records = new ExternalSpillableMap<>(maxMemorySizeInBytes/2, readerSchema,
```

Comment:
@vinothchandar here, I guess we can changes this to `maxMemorySizeInBytes` since we will use only the in memory Maps.newHashMap() below ? The problem is We need account for Maps.newHashMap() since we are still reading block level ?

Reply:
we need to budget for size of a data block correct (after we split it into pieces) . but we can remove the /2 here. sure

Reply:
sg.